helpers.ts

```diff
@@ -337,6 +338,16 @@ export function isPromise(candidateFuture: any): boolean {
 	return !!(candidateFuture && typeof (candidateFuture.then) === "function");
 }
 
+export async function attachAwaitDetach(eventName: string, ee: EventEmitter, cb: Function, operation: Promise<any>) {
```

Comment:
for readability I wouldn't name the variables like this in an exported function - ee/cb